tardis/widgets/line_info.py

```diff
@@ -0,0 +1,749 @@
+            col_widths=line_counts_col_widths,
+            table_options=max_rows_option,
+        )
+        self.total_packets_label = TableSummaryLabel(
+            target_table=self.last_line_counts_table,
+            table_col_widths=line_counts_col_widths,
+            label_key="Total Packets",
+            label_value=0,
+        )
+
+        self.figure_widget = self.plot_spectrum(
+            spectrum_wavelength,
+            spectrum_luminosity_density_lambda,
+            virt_spectrum_wavelength,
+            virt_spectrum_luminosity_density_lambda,
+        )
+
+        self.filter_mode_buttons = ipw.ToggleButtons(
+            options=self.FILTER_MODES_DESC, index=0
+        )
+
+        self.group_mode_dropdown = ipw.Dropdown(
+            options=self.GROUP_MODES_DESC, index=0
+        )
+
+    @classmethod
+    def from_simulation(cls, sim):
+        """
+        Create an instance of LineInfoWidget from a TARDIS simulation object.
+
+        Parameters
+        ----------
+        sim : tardis.simulation.Simulation
+            TARDIS Simulation object produced by running a simulation
+
+        Returns
+        -------
+        LineInfoWidget object
+        """
+        return cls(
+            lines_data=sim.plasma.lines.reset_index().set_index("line_id"),
+            line_interaction_analysis={
+                filter_mode: LastLineInteraction.from_model(sim, filter_mode)
+                for filter_mode in cls.FILTER_MODES
+            },
+            spectrum_wavelength=sim.runner.spectrum.wavelength,
+            spectrum_luminosity_density_lambda=sim.runner.spectrum.luminosity_density_lambda,
+            virt_spectrum_wavelength=sim.runner.spectrum_virtual.wavelength,
+            virt_spectrum_luminosity_density_lambda=sim.runner.spectrum_virtual.luminosity_density_lambda,
+        )
+
+    def get_species_interactions(
+        self, wavelength_range, filter_mode=FILTER_MODES[0]
+    ):
+        """
+        Get fractional species interactions in specified wavelength range.
+
+        Fractional species interactions means fraction of packets present in
+        the specified wavelength range which experienced their last interaction
+        with a species. The packets to consider are filtered by the specified
+        filter mode.
+
+        Parameters
+        ----------
+        wavelength_range : list-like or None
+            A list of two float values to specify the wavelength range - first 
+            for the range start and second for the range end. None specifies
+            that no wavelength range is selected and will return empty dataframe
+        filter_mode : str, optional
+            Filter mode of the LastLineInteraction object to use for filtering
+            the selected wavelength range (more details in Notes section).
+            Allowed values are given by the class variable :code:`FILTER_MODES`
+            (default value is :code:`FILTER_MODES[0]`)
```

Comment:
Should probably specify here what FILTER_MODES[0] actually is

Reply:
Yes I agree with @ycamacho and in built api docs, the class properties are also enlisted so it will be clear for readers!